src/lib/CLValue2/Result.ts

```diff
@@ -0,0 +1,53 @@
+import { concat } from '@ethersproject/bytes';
+import { Result, Ok, Err} from "ts-results";
+
+import { CLValue, CLType, ToBytes } from './index';
+import { toBytesU8 } from '../ByteConverters';
+
+export enum CLErrorCodes {
+  EarlyEndOfStream = 0,
+  Formatting,
+  LeftOverBytes,
+  OutOfMemory
+}
+
+const RESULT_TAG_ERROR = 1;
+const RESULT_TAG_OK = 1;
+
+export class CLResultType extends CLType {
+  toString(): string {
+    return 'Result';
+  }
+}
+
+export class GenericResult<T, E> {
```

Comment:
Not sure do we need this `GenericResult` but IMHO its handy to extends into CLResult as we still have possibility to create different internal Result types.